src/PHPePub/Core/EPub.php

```diff
@@ -1457,7 +1457,7 @@ function getLanguage() {
      * @return bool $success
      */
     function setLanguage($language) {
-        if ($this->isFinalized || mb_strlen($language) != 2) {
+        if ($this->isFinalized || mb_strlen($language) < 2 || mb_strlen($language) > 5) {
```

Comment:
What about values like `zh-hant` ?

Reply:
https://stackoverflow.com/a/17863380

Reply:
Also `_` is not supported, separator is `-`

Reply:
We can validate the locale code if we require Intl https://www.php.net/manual/en/resourcebundle.locales.php

Reply:
So lower than 2 and greater than 7 or 8 ?

Reply:
Do we really need a complex validation for the language?

Reply:
You could replace mr_strlen calls with this regex:

``` php
preg_match('/^((?<language>([A-Za-z]{2,3}(-(?<extlang>[A-Za-z]{3}(-[A-Za-z]{3}){0,2}))?)|[A-Za-z]{4}|[A-Za-z]{5,8})(-(?<region>[A-Za-z]{2}|[0-9]{3}))?)$/', $language);
```

It is incomplete but should work for almost all cases.

References:
- https://github.com/sebinsua/ietf-language-tag-regex/blob/master/index.js
- https://schneegans.de/lv/#regex

Reply:
Note:

This regex will accept `en`, `fr`, `fr-FR`, … while refusing things like `fr-`.